apps/files_sharing/src/components/NewFileRequestDialog/NewFileRequestDialogFinish.vue

```diff
@@ -0,0 +1,224 @@
+<!--
+  - SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div>
+		<!-- Request note -->
+		<NcNoteCard type="success">
+			{{ t('files_sharing', 'Once created, you can share the link below to allow people to upload files to your directory.') }}
+		</NcNoteCard>
+
+		<!-- Copy share link -->
+		<NcInputField ref="clipboard"
+			:value="shareLink"
+			:label="t('files_sharing', 'Share link')"
+			:readonly="true"
+			:show-trailing-button="true"
+			:trailing-button-label="t('files_sharing', 'Copy to clipboard')"
+			@click="copyShareLink"
+			@click-trailing-button="copyShareLink">
+			<template #trailing-button-icon>
+				<IconCheck v-if="isCopied" :size="20" @click="isCopied = false" />
+				<IconClipboard v-else :size="20" @click="copyShareLink" />
```

Comment:
```suggestion
				<IconCheck v-if="isCopied" :size="20" />
				<IconClipboard v-else :size="20" />
```

No interaction listeners on non interactive elements.
This should be done in `copyShareLink`